core/src/main/java/org/apache/accumulo/core/spi/balancer/TableLoadBalancer.java

```diff
@@ -23,21 +23,38 @@
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
 
+import org.apache.accumulo.core.Constants;
 import org.apache.accumulo.core.classloader.ClassLoaderUtil;
 import org.apache.accumulo.core.conf.Property;
 import org.apache.accumulo.core.data.TableId;
 import org.apache.accumulo.core.data.TabletId;
 import org.apache.accumulo.core.manager.balancer.AssignmentParamsImpl;
 import org.apache.accumulo.core.manager.balancer.BalanceParamsImpl;
+import org.apache.accumulo.core.spi.balancer.data.TServerStatus;
 import org.apache.accumulo.core.spi.balancer.data.TabletMigration;
 import org.apache.accumulo.core.spi.balancer.data.TabletServerId;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
+ * TabletBalancer that balances Tablets for a Table using the TabletBalancer defined by
+ * {@link Property#TABLE_LOAD_BALANCER}. This allows for different Tables to specify different
+ * TabletBalancer classes.
+ * <p>
+ * Note that in versions prior to 4.0 this class would pass all known TabletServers to the Table
+ * load balancers. In version 4.0 this changed with the introduction of
+ * {@link Property#TABLE_ASSIGNMENT_GROUP} such that this balancer now only passes the TabletServers
```

Comment:
If using the table.custom prop these docs could have text like : 

```
Set table.custom.assignment.group=<group name> to limit assignment of tablets to tablet servers that were started with the named group.   
```

Reply:
I moved the new property into TableLoadBalancer, and it's named `table.custom.assignment.group`. I updated the javadoc.